src/fsharp/TastOps.fs

```diff
@@ -3004,7 +3004,22 @@ let isByrefLikeTy g m ty =
 
 let isSpanLikeTy g m ty =
     isByrefLikeTy g m ty && 
-    not (isByrefTy g ty) 
+    not (isByrefTy g ty)
+
+let isSpanTyconRef (g: TcGlobals) m (tcref: TyconRef) =
+    if isByrefLikeTyconRef g m tcref then
+        match tcref.ILTyconInfo with
+        | TILObjectReprData(_, _, ilTypeDef) -> ilTypeDef.Name = "System.Span`1"
+    else
+        false
+    
+let isSpanTy g m ty =
+    ty |> stripTyEqns g |> (function TType_app(tcref, _) -> isSpanTyconRef g m tcref | _ -> false)
+
+let destSpanTy (g:TcGlobals) m ty =
```

Comment:
Instead of isdestspanty and destspanty consider to put it in one function that returns valueoption. This removes that weird failwith and also does the stripping only once

Reply:
@forki This is just following existing structure in TastOps.fs, I think it's ok and proper